vmware-tanzu/tanzu-framework
# Decouple Feature-Flags from `cli/runtime` library and move to it to plugin specific code

### What this PR does / why we need it

- This PR decouples feature-flag definitions from the `cli/runtime` library and move it to plugin specific code
- It also uses `postInstallHook` for a specific plugin to set the defaults for the feature-flags in the config yaml

### Which issue(s) this PR fixes


Fixes #3447

### Describe testing done for PR
- **Manual tests:**
  - Clean tanzu config file
  - Run `tanzu config get` command
    - It only adds tanzu cli specific default feature-flags to the config
  - Install all plugins with `tanzu plugin sync` to install all plugins
    -  (As part of plugin installation `postInstallHook` should configure the plugin specific feature-flags)
    - Run `tanzu config get` and verify plugin specific default feature-flags are configured correctly.
- **Unit tests** are written and validated as part of CI 



### Release note

```release-note
Decouple Feature-Flags from `cli/runtime` library and move to it to plugin specific code
```



### Additional information

#### Special notes for your reviewer
